modules/apps/transfer/keeper/relay.go

```diff
@@ -201,7 +201,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		// NOTE: We use SourcePort and SourceChannel here, because the counterparty
 		// chain would have prefixed with DestPort and DestChannel when originally
 		// receiving this coin as seen in the "sender chain is the source" condition.
-		if token.Denom.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel()) {
+		if token.Denom.HasPrefix(packet.GetSourcePort(), packet.GetSourceChannel()) {
```

Comment:
I personally don't feel like this reads easier, but I'm okay with the change. I think either restructing the comments or flip flopping the if statements could make it better, but as is requires me to reparse all the logic to figure out which chain is source and which port and channel to be passing in

Reply:
maybe you could update the comment:
```go
// if the token we are receiving has the packet source port and channel prefixed
// then the sending chain must have prefixed this hop to the token when it originally
// received these tokens from us. Given that we are receiving back the tokens
// we should remove the prefix added by the counterparty and unescrow the tokens. 
```

I think the previous comment is more relevant to the previous function and even references it ""sender chain is source" condition"